mypy/plugins/dataclasses.py

```diff
@@ -965,6 +965,9 @@ def dataclass_tag_callback(ctx: ClassDefContext) -> None:
 
 def dataclass_class_maker_callback(ctx: ClassDefContext) -> bool:
     """Hooks into the class typechecking process to add support for dataclasses."""
+    if any(i.is_named_tuple for i in ctx.cls.info.mro):
+        ctx.api.fail("A NamedTuple cannot be a dataclass.", ctx=ctx.cls.info)
```

Comment:
```suggestion
        ctx.api.fail("A NamedTuple cannot be a dataclass", ctx=ctx.cls.info)
```

small nitpick, everything else LGTM :)

Reply:
Yes, you're right. Thanks!